### ISSUE TITLE
[Feature Request] Unnamed values

### ISSUE THREAD

**Issue by Apeiros-46B** [jbyuki/quickmath.nvim#4]
It could be quite useful if unnamed values were supported, i.e., being able to enter an expression without assigning it to a variable in the buffer (e.g. `1 + 2`) and have it calculate instead of returning an error.

Currently, you need to enter `somevariable = 1 + 2` instead of just `1 + 2`, which I think is restricting when working with temporary values that don't need names.

**Current behaviour:**
![image](https://user-images.githubusercontent.com/61390942/201539030-4997ff32-f8fa-4859-a177-db3d8b67e98f.png)

**Requested behaviour:** (rough mockup)
![image](https://user-images.githubusercontent.com/61390942/201539150-cf240086-da20-4c5e-950d-13a689b75c56.png)

**Comment by jbyuki** [comment]
That could be a good idea, quickmath is fairly basic now so these kind of improvements could help it make a better tool. I can see a line per line execution to support this feature. 
If you're willing, you can try to implement it. The core code is short and if you know some Lua, it shouldn't be hard. I'll be willing to review an implementation.

**Comment by Apeiros-46B** [comment]
I'll see what I can do. I'm not the best at Lua, but I think I can grasp what the code does.  
I have no idea how to use `ntangle` though

**Comment by jbyuki** [comment]
If you can use ntangle it's great, otherwise just modifying the lua code is fine as well. I can modify the rest.
Don't hestitate to ask if you need to know anything.

**Comment by Apeiros-46B** [comment]
I've made some small progress  
Basic operators with scalars work, but there are still some issues in other places  
Also haven't tested it that much but I don't think anything else broke (Will test more later though)
I'll make a PR when I think this is done  
![image](https://user-images.githubusercontent.com/61390942/201552922-a29f6310-8674-4a60-a265-f3a19f438778.png)

**Comment by jbyuki** [comment]
Nice, looking good

**Comment by Apeiros-46B** [comment]
Currently calculation all works properly and I fixed errors not showing, but the virtual text seems to be really glitchy as a side effect of my changes
I'm trying to figure out how to attach a callback to when lines are added or deleted so I can shift the line numbers accordingly, any ideas?
Also, should we move this discussion to a PR or continue it here?

https://user-images.githubusercontent.com/61390942/201565231-75121f7f-dbb5-44de-b22f-97b6bd4a6e13.mp4

**Comment by jbyuki** [comment]
Yes, we can move this to a PR.

**Comment by Apeiros-46B** [comment]
After a few attempts, I've realized that this is probably going to take more of my time than I expected, so I might do it at some later date
For now, I've pushed my current implementation to [my fork](https://github.com/Apeiros-46B/quickmath.nvim) if you want to take a look (Not going to make a PR since it's still in a half-broken state)
Also not sure about my code quality, so I think it might be better if you implement it if you're willing

**Comment by jbyuki** [comment]
I had a look in the code. Does this solve the issue with empty lines?

```diff
index 2f73b02..593f61a 100644
--- a/lua/quickmath/init.lua
+++ b/lua/quickmath/init.lua
@@ -54,6 +54,7 @@ local function StartSession()
             local anon = false
             local f, success, errmsg
 
+		if line ~= "" then
 			if string.find(line, "^%w+%s*=") then
 				name = string.match(line, "^(%w+)%s*=")
 
@@ -70,6 +71,7 @@ local function StartSession()
             if success and anon then
                 anon_n = anon_n + 1
             end
+	end
 		end
```

You can go ahead and open a PR anyway even if it's in a broken state (draft PR or normal PR). I can try to fix the last remaining parts. The overall implementation looks good already.

**Comment by Apeiros-46B** [comment]
Virtual text still seems to be broken/not shifted properly when a new line is inserted
I'll make a PR for now and you can continue it

**Comment by jbyuki** [comment]
Closed with #5
